### ISSUE TITLE
A question regarding render times and resuming after process crash

### ISSUE THREAD

**Issue by Delta-38** [overviewer/Minecraft-Overviewer#1859]
Hi, 
I'm running Overviewer (not the MS Store version) to render a relatively large world (around 1GB) but have been facing a few issues. 
1) Despite running on relatively modern hardware (i7-6700K 16GB ram, SSD and whatnot, albeit on windows) even a single smooth_lighting render of the overworld in a single direction requires at least 1.5hours, and a half-day change update requires about 45 minutes. This isn't an issue per se. However, given I would like to render the 4 directions for day, one for night, one for nether and end ... It turns out to a very very long render (more than a day long) at about 10-15t/s, whereas the server machine the map should be deployed on, an i3, only manages about 0.8 1.4 t/s so it's much too slow. 

I have tried making it an incremental process, as in ... Config file to render 1 mode, then when done, second file etcetera... 
however it appears the rendering of multiple modes is simultaneous, meaning that if in this lenghty process, there is any kind of glitch and the process has to be restarted, all selected renders get selected for force rendering. 

Is there any way to split/merge or make the production of this first megarender more predictable? 

I had successfully completed it once (27hours, 34GB folder), but during one of the following updates a prolonged power outage caused an interruption, and of course on following restart a force render was called again. 

Is there any way to manage this? or to wrap the process in a stop/start kind of way?

I love overviewer but it is really giving me a headache this time around.

TL;DR; 
Is there a consistent strategy to produce a large set of renders, or to come back from an error in update without rerendering each and every map, even if unchaged?
Can the process be optimized in any way (I didn't really see appreciable difference in render time between normal, smooth and optipng settings) (IDK maybe on linux it runs at three times the speed?)

Thanks in advance, 
Paolo Manili

**Comment by cliffmeyers** [comment]
Just wanted to check what value of `processes` you're using in your config file, or via the `-p` arg?

**Comment by Delta-38** [comment]
I have tried both auto (looking into the process it goes to 8 processes so one per thread) and leaving it at 2-4 (slows down accordingly). 
I have just run again a first render of the overworld in a single direction, smoothlighting, no further optimization. 
`2020-10-10 11:30:45  Welcome to Minecraft Overviewer version 0.16.5 (01ed588)!
2020-10-10 11:30:46  Generating textures...
2020-10-10 11:31:29  Preprocessing...
2020-10-10 11:34:26  Rendering 101456 total tiles.
 99% [======================================= ] 101433 16.97t/s eta 00h 00m 01s
2020-10-10 13:14:03  Rendering complete!`

This isn't per se too slow (Despite requiring dedicating the most powerful machine in the house to the task doing nothing else), however the 4 directions plus nether and end turn to more than a day of runtime, and it has happened more than once to me that for odd reasons it fails to complete or stays stuck towards the end. 

I would appreciate it even if there was a way to selectively update a single render so as not to risk a force-render or tilecheck all over, but it is unclear to me how this deals with the index and menus.

**Comment by Delta-38** [comment]
To post an update: 
After leaving the big machine alone for a weekend generating a new clean compressed png map of the overworld, I've enjoyed relative success in leaving the minecraft server to render updates on a low priority process. 
Despite a day of save delta requiring nearly a day of processing (2threads on an oldish i3, same overviewer configuration as first render, 30K tile updates). 
Today Windows (I know, I'm cursed with the server os I chose) decided to restart, killing off the Powershell that I invoked the rendering from. 
Now, of course, Overviewer detects the rendering as having been interrupted and will have to check all tiles. 
Is there any way to make it treat like a simple update? 

This is an excerpt from rendering updates to provide some data on the performance: 
processes=1 i3-3250
2020-10-14 18:56:00  Welcome to Minecraft Overviewer version 0.16.5 (01ed588)!
2020-10-14 18:56:01  Generating textures...
2020-10-14 18:56:24  Generating textures...
2020-10-14 18:56:44  Generating textures...
2020-10-14 18:57:04  Generating textures...
2020-10-14 18:57:27  Preprocessing...
2020-10-14 19:01:35  Rendering 28607 total tiles.
 99% [========================================= ] 28600 0.24t/s eta 00h 00m 29s
2020-10-16 05:09:34  Rendering complete!
2020-10-16 05:09:35  Your render has been written to 'path', open index.html to view it.

processes=2 i3-3250
PS path> .\overviewer.exe -c overconf.conf
2020-10-16 08:33:57  Welcome to Minecraft Overviewer version 0.16.5 (01ed588)!
2020-10-16 08:34:00  Generating textures...
2020-10-16 08:34:18  Generating textures...
2020-10-16 08:34:39  Generating textures...
2020-10-16 08:35:00  Generating textures...
2020-10-16 08:35:19  Preprocessing...
2020-10-16 08:40:25  Rendering 29425 total tiles.
 99% [========================================= ] 29400 0.51t/s eta 00h 00m 49s
2020-10-17 00:50:47  Rendering complete!
